vault/plugin_reload.go

```diff
@@ -198,6 +198,15 @@ func (c *Core) reloadBackendCommon(ctx context.Context, entry *MountEntry, isAut
 	re.backend = backend
 
 	if backend != nil {
+		// Initialize the backend after reload. This is a no-op for backends < v5 which
+		// rely on lazy loading for initialization. v5 backends do not rely on lazy loading
+		// for initialization unless the plugin process is killed. Reload of a v5 backend
+		// results in a new plugin process, so we must initialize the backend here.
+		err := backend.Initialize(c.activeContext, &logical.InitializationRequest{Storage: view})
```

Comment:
I'm unsure if this should use the request context or the core's activeContext. I used activeContext because that's what's used in [auth.go](https://github.com/hashicorp/vault/blob/main/vault/auth.go#L229-L232) and [mount.go](https://github.com/hashicorp/vault/blob/main/vault/mount.go#L673-L676). I'm don't totally understand why it's used in those files either.

Reply:
We'll use the request context here. Changing this.

Reply:
Done in https://github.com/hashicorp/vault/pull/17248/commits/3e696dc496adf9d88424eb1f043bff2fb724de42.